syncd/VirtualOidTranslator.cpp

```diff
@@ -22,6 +22,28 @@ VirtualOidTranslator::VirtualOidTranslator(
     // empty
 }
 
+sai_object_id_t VirtualOidTranslator::tryTranslateRidToVid(
+        _In_ sai_object_id_t rid,
+        _In_ sai_object_id_t switchVid)
+{
+    SWSS_LOG_ENTER();
+
+    /*
+     * NOTE: switch_vid here is Virtual ID of switch for which we need
+     *  get VID for given RID. No check now! 
+     */
+
+    if(false == checkRidExists(rid))
```

Comment:
this entire method must be under mutex, since there can be context switch between checkRidExists and getVidForRid
also, trytranslate should return bool if success, and it assumes that will not create new vid if rid exists, so switchVid is not necessary param here, you are not even using it

Reply:
space after if

Reply:
ok. I removed the lock since checkRidExists() already has it and  getVidForRid is DB call.  I can add it.
Will remove switchVid  and introduce  out parameter for vid.
Will add space after if.

Please confirm.

Reply:
need lock to get the right data what we checked for.

Reply:
checkRidExists uses mutex internally, so you cant use this method directly when you will add mutex here, it would be better to use checkRidExists contents here